client/my-sites/plugins/index.js

```diff
@@ -35,6 +36,16 @@ module.exports = function() {
 	}
 
 	if ( config.isEnabled( 'manage/plugins' ) ) {
+		page( '/plugins/wpcom-masterbar-redirect/:site', context => {
```

Comment:
I tried it with `context.query` and this seems simpler to me. If I'm not missing something, in that case I'd have to move the redirect logic here https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins/controller.js#L197, and that seems messier to handle.